firmware/CMakeLists.txt

```diff
@@ -43,6 +43,9 @@ if(NOT DEFINED ${CMAKE_CXX_COMPILER_VERSION})
 	string(STRIP ${CMAKE_CXX_COMPILER_VERSION} CMAKE_CXX_COMPILER_VERSION)
 endif()
 
+# Set EXPECTED_GCC_VERSION to match current toolchain
+set(EXPECTED_GCC_VERSION "10.3.1")
```

Comment:
Why change the gcc version ?
It's known to bring problems everywhere else and it would need a dedicated PR and tests just for it.